src/Components/ObjectProperties/ObjectProperties.js

```diff
@@ -57,6 +59,12 @@ class ObjectProperties extends React.Component {
     this.subscription = this.context.eventEmitter.addListener('alignKeys', this.alignKeys);
   };
 
+  componentWillReceiveProps = (nextProps) => {
+    if (nextProps.keyWidth) {
+      this.setState({ keyWidth: nextProps.keyWidth });
```

Comment:
Ok I verified on React documentation - it seems perfectly fine to call `setState` in this lifecycle method.